server/coordinator/scheduler/assign_shard_scheduler.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package scheduler
+
+import (
+	"context"
+
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/CeresDB/ceresmeta/server/coordinator"
+	"github.com/CeresDB/ceresmeta/server/storage"
+)
+
+const (
+	AssignReason = "ShardView exists in metadata but shardNode not exists, assign shard to node"
+)
+
+// AssignShardScheduler used to assigning shards without nodes.
+type AssignShardScheduler struct {
+	factory    *coordinator.Factory
+	nodePicker coordinator.NodePicker
+}
+
+func NewAssignShardScheduler(factory *coordinator.Factory, nodePicker coordinator.NodePicker) Scheduler {
+	return &AssignShardScheduler{
+		factory:    factory,
+		nodePicker: nodePicker,
+	}
+}
+
+func (a AssignShardScheduler) Schedule(ctx context.Context, topology cluster.Topology) (Result, error) {
+	if topology.ClusterView.State != storage.ClusterStateStable {
+		return Result{}, cluster.ErrClusterStateInvalid
+	}
+
+	// Check if there is a shard without node mapping.
+	for _, shardView := range topology.ShardViews {
+		exists, _ := contains(shardView.ShardID, topology.ClusterView.ShardNodes)
```

Comment:
Maybe we can avoid copy caused by the loop:
```suggestion
	for idx := range topology.ShardViews {
	        shardID := topology.ShardViews[idx].ShardID
		exists, _ := contains(shardView.ShardID, topology.ClusterView.ShardNodes)
```